### ISSUE TITLE
Call after methods in axe.runVirtualRule

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4037]
Because `axe.runVirtualRule` isn't calling `after` methods, it can't give correct results for rules that use them. I'm not aware of anyone who's using runVirtualRule using it to run rules with an after method. It would've been reported otherwise, but someone might.

This should result in a violation:

```js
var node = new axe.SerialVirtualNode({ nodeName: 'div'});
node.children = [
  new axe.SerialVirtualNode({ nodeName: 'main'}),
  new axe.SerialVirtualNode({ nodeName: 'main'})
];
var results = axe.runVirtualRule('landmark-no-duplicate-main', node);
```
